package/android/cpp/rnskia-android/SkiaOpenGLSurfaceFactory.h

```diff
@@ -42,12 +44,34 @@ class ThreadContextHolder {
  */
 class WindowSurfaceHolder {
 public:
-  WindowSurfaceHolder(jobject surface, int width, int height)
-      : _width(width), _height(height),
-        _window(ANativeWindow_fromSurface(facebook::jni::Environment::current(),
-                                          surface)) {}
+  WindowSurfaceHolder(jobject jSurfaceTexture, int width, int height)
```

Comment:
This looks a bit "cluttered" in the surface holder class - could we instead change the jSurfaceTexture parameter to be a JNI object implemented outside of this method to clean up this a bit? 

- Let the SkiaBaseView pass a java object in `onSurfaceAvailable` that contains both the texture and the callback for calling updateTexImage
- Implement this object as a JNI object that hides all the manual JNI code below?

Reply:
This is a good idea. Maybe not worth it if we change model in the mid-term (rendering thread?).

Reply:
to this point, I actually have a branch where even all the opengl calls are also in java (`df` branch) so I am definitely a believer of this approach. Skottielib does this well too (only the C++ calls are done in C++).